app/src/main/java/com/github/se/polyfit/di/UserModule.kt

```diff
@@ -0,0 +1,44 @@
+package com.github.se.polyfit.di
+
+import android.content.Context
+import com.github.se.polyfit.data.remote.firebase.MealFirebaseRepository
+import com.github.se.polyfit.model.data.User
+import com.github.se.polyfit.ui.utils.AuthenticationCloud
+import com.github.se.polyfit.viewmodel.meal.MealViewModel
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object UserModule {
+
+  @Provides
+  @Singleton
+  fun providesUser(): User {
+    return User(id = "test")
```

Comment:
This is primarily to avoid a bug when testing separate components of the app without going through the login screen. When a user logs in, their user ID is stored and later used in MealFirebase. If no ID is set, it causes an error.

Reply:
I see, makes sense but I do wonder if its the best way to do it? I'm not too familiar in the domain so if you think it is then no issue

Reply:
I removed the default ID for the user and changed the navigation to start on the home screen. Now, I get this error:

```java
java.lang.IllegalArgumentException: Invalid document reference. Document references must have an even number of segments, but 'users' has 1.


In theory, we should never be able to reach this point. It might be a good idea to keep it as a safeguard to prevent the app from crashing. However, it could also be the source of potential bugs in the future—for example, user data being stored under the wrong user. What do you think?

Reply:
o.O

Damn. Yeah, I think its a good safeguard. Just leave a comment about it in the code & lets mention it to the whole team at standup so if anyone notices weird stuff they might have an idea